lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -279,59 +291,47 @@ module.exports = {
         cloudwatchLogEvent.logSubscriptionSerialNumber
       );
 
-      return {
-        destinationArn,
-        logicalId,
-      };
+      return { destinationArn, logicalId };
     });
 
-    const sameDestinationArn = (sf1, sf2) => sf1.destinationArn === sf2.destinationArn;
-    const sameLogicalId = (sf1, sf2) => sf1.logicalId === sf2.logicalId;
-
-    // Old subscription filters which don't have the same destination as any of the new subscription filters.
-    // Currently, these kinds of subscription filters are deleted.
-    // In the future, the behavior will change to not delete filters that were set up externally.
-    const notMatchedOldSubscriptionFilters = oldSubscriptionFilters.filter(
-      (oldSubscriptionFilter) => {
-        const destinationMatchNewSubscriptionFilter = newSubscriptionFilters.find(
-          (newSubscriptionFilter) =>
-            sameDestinationArn(newSubscriptionFilter, oldSubscriptionFilter)
-        );
-        return !destinationMatchNewSubscriptionFilter;
-      }
+    // Deleting external defined subscription filter is dangerous.
```

Comment:
Let's rephrase that comment as:

```
// If subscription filters defined externally cause a situation where we cannot create all 
// subscription filters defined as a part of current service, we want to throw an error
// instead of silently removing external filters.
```